src/main/java/com/project/workaholic/vcs/service/OAuthService.java

```diff
@@ -13,25 +13,32 @@
 @Service
 @RequiredArgsConstructor
 public class OAuthService {
-    private final OAuthGithubService githubService;
+    private final GithubService githubService;
+    private final GitlabService gitlabService;
     private final OAuthAccessTokenRepository tokenRepository;
 
     public RedirectView importVCS(RedirectAttributes redirectAttributes, VCSVendor vendor) {
-        switch (vendor) {
-            case GITHUB :
-                    return githubService.requestCode(redirectAttributes);
-            default:
-                throw new CustomException(StatusCode.INVALID_VCS_VENDOR);
-        }
+        return switch (vendor) {
+            case GITHUB -> githubService.requestCode(redirectAttributes);
+            case GITLAB -> gitlabService.requestCode(redirectAttributes);
+        };
```

Comment:
```suggestion
        return vcsServices.get(vendor).requestCode(redirectAttributes);
```
그냥 든 생각인데, 이거야 말로, githubService와 gitlabService를 vcsService정도의 공통 인터페이스로 빼서, 정도로 할 수 있지 않았을까 하는 ... 욕심이 ... 드는데 말이죠.. :-)

Reply:
물론 이렇게 하면, 나중에 방식이 다른 vcs가 추가되었을때, 코드가 아주 지저분해지는 단점이 있습니다. ㅎㅎ